optax/_src/transform.py

```diff
@@ -1201,15 +1206,30 @@ def scale_by_optimistic_gradient(
   """
 
   def init_fn(params):
-    return TraceState(trace=otu.tree_zeros_like(params))
+    return ScaleByOptimisticGradientState(
+        is_initial_step=jnp.array(True),
+        previous_gradient=otu.tree_zeros_like(params),
+    )
 
   def update_fn(updates, state, params=None):
     del params
 
     new_updates = jax.tree.map(
-        lambda grad_t, grad_tm1: (alpha + beta) * grad_t - beta * grad_tm1,
-        updates, state.trace)
-    return new_updates, TraceState(trace=updates)
+        lambda grad_t, grad_tm1: jnp.where(
```

Comment:
could you please add a comment explaining the initial step issue and why the jnp.where is necessary ?

Reply:
@fabianp Done.